### PR TITLE
adds new azure sub command structure

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/3566.

### CODE REVIEW

File: cmd/aws.go [hunk 1]
```diff
@@ -17,7 +14,7 @@ import (
 var (
 	AwsCmd = &cobra.Command{
 		Use:   "aws",
-		Short: "cleanup leftover AWS CI resources",
+		Short: "Cleanup leftover AWS CI resources.",
```
Comment (xh3b4sd) [inline cmd/aws.go]: Boyscouting. The help usages are sentences everywhere.

File: cmd/aws.go [hunk 2]
```diff
@@ -26,18 +23,9 @@ var (
 	accessKeyID     string
 	secretAccessKey string
 	region          string
-	logger          *micrologger.MicroLogger
```
Comment (xh3b4sd) [inline cmd/aws.go]: The logger is now in the `root.go`. I like the more structured project layout in `opsctl` more than this here but I am not going to rewrite everything now.

File: cmd/azure.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/cobra"
+
+	pkgazure "github.com/giantswarm/ci-cleaner/pkg/cleaner/azure"
+)
+
+var (
+	AzureCmd = &cobra.Command{
```
Comment (xh3b4sd) [inline cmd/azure.go]: New azure sub command.

File: pkg/cleaner/aws/aws.go [hunk 4]
```diff
@@ -12,12 +12,12 @@ import (
 
 type Config struct {
 	CFClient CFClient
-	Logger   *micrologger.MicroLogger
+	Logger   micrologger.Logger
```
Comment (xh3b4sd) [inline pkg/cleaner/aws/aws.go]: Boyscouting. This is the interface.

File: pkg/cleaner/azure/cleaner.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+	return c, nil
+}
+
+func (c *Cleaner) Clean(ctx context.Context) error {
+	// TODO cleanup peeringso
+	//
+	//     https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-05-01/network#VirtualNetworksClient
+	//     https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-05-01/network#VirtualNetworkPeeringsClient
+	//
```
Comment (xh3b4sd) [inline pkg/cleaner/azure/cleaner.go]: Comes next.

**Review by tuommaki — APPROVED** [review]
I'm not an Azure expert, but in general this looks good.

**Pull request merged**